hail_search/hail_search_query.py

```diff
@@ -998,23 +1029,133 @@ def _format_results(self, ht, annotation_fields):
         ht = ht.annotate(selected_transcript=self._selected_main_transcript_expr(ht))
         return super()._format_results(ht, annotation_fields)
 
-    @staticmethod
-    def _gene_ids_expr(ht):
-        return hl.set(ht.sorted_transcript_consequences.map(lambda t: t.gene_id))
-
     @classmethod
     def _omim_sort(cls, r, omim_gene_set):
         return [
             hl.if_else(omim_gene_set.contains(r.selected_transcript.gene_id), 0, 1),
-            -cls._gene_ids_expr(r).intersection(omim_gene_set).size(),
-        ]
+        ] + super()._omim_sort(r, omim_gene_set)
 
     @classmethod
     def _gene_rank_sort(cls, r, gene_ranks):
-        return [
-            gene_ranks.get(r.selected_transcript.gene_id),
-            hl.min(cls._gene_ids_expr(r).map(gene_ranks.get)),
-        ]
+        return [gene_ranks.get(r.selected_transcript.gene_id)] + super()._gene_rank_sort(r, gene_ranks)
+
+
+class SvHailTableQuery(BaseHailTableQuery):
+
+    DATA_TYPE = 'SV_WGS'
+
+    GENOTYPE_FIELDS = {_to_camel_case(f): f for f in ['CN', 'GQ']}
+    NESTED_GENOTYPE_FIELDS = {'concordance': ['new_call', 'prev_call', 'prev_num_alt']}
+    GENOTYPE_QUERY_FIELDS = {'gq_sv': 'GQ', 'gq': None}
+
+    TRANSCRIPTS_FIELD = 'sorted_gene_consequences'
+    TRANSCRIPT_CONSEQUENCE_FIELD = 'major_consequence'
+    CORE_FIELDS = BaseHailTableQuery.CORE_FIELDS + ['algorithms']
+    BASE_ANNOTATION_FIELDS = {
+        'bothsidesSupport': lambda r: r.bothsides_support,
+        'chrom': lambda r: r.start_locus.contig.replace('^chr', ''),
+        'endChrom': lambda r: hl.or_missing(r.start_locus.contig != r.end_locus.contig, r.end_locus.contig.replace('^chr', '')),
+        'pos': lambda r: r.start_locus.position,
+        'end': lambda r: r.end_locus.position,
+        'rg37LocusEnd': lambda r: r.rg37_locus_end,
+    }
+    BASE_ANNOTATION_FIELDS.update(BaseHailTableQuery.BASE_ANNOTATION_FIELDS)
+    ENUM_ANNOTATION_FIELDS = {
+        TRANSCRIPTS_FIELD: BaseHailTableQuery.ENUM_ANNOTATION_FIELDS['transcripts'],
+    }
+
+    POPULATIONS = {
+        'sv_callset': {'hemi': None, 'sort': 'callset_af'},
+        'gnomad_svs': {'id': 'ID', 'ac': None, 'an': None, 'hom': None, 'hemi': None, 'het': None, 'sort': 'gnomad'},
+    }
+    POPULATION_FIELDS = {'sv_callset': 'gt_stats'}
+    PREDICTION_FIELDS_CONFIG = {
+        'strvctvre': PredictionPath('strvctvre', 'score'),
+    }
+
+    SORTS = {
+        'protein_consequence': lambda r: [hl.min(r.sorted_gene_consequences.map(lambda g: g.major_consequence_id))],
```

Comment:
Use `r[TRANSCRIPTS_FIELD]` for `r.sorted_gene_consequences`?

Reply:
its not in scope here so it would have to be `r[SvHailTableQuery. TRANSCRIPTS_FIELD]` which is less readable. The purpose of that varianble is to support methods in the base class, not so much to use it every possibel location